src/libraries/Common/tests/System/Security/Cryptography/MLKemCngTests.Windows.cs

```diff
@@ -0,0 +1,209 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+
+using Xunit;
+using Xunit.Sdk;
+
+using KeyBlobMagicNumber = Interop.BCrypt.KeyBlobMagicNumber;
+
+namespace System.Security.Cryptography.Tests
+{
+    [ConditionalClass(typeof(MLKem), nameof(MLKem.IsSupported))]
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public sealed class MLKemCngPlaintextExportableTests : MLKemCngTests
+    {
+        protected override CngExportPolicies ExportPolicies => CngExportPolicies.AllowExport | CngExportPolicies.AllowPlaintextExport;
+    }
+
+    // ML-KEM as of Windows build 27881 does not have PKCS#8 exports, so we cannot implement encrypted exports.
+    [ActiveIssue("https://github.com/dotnet/runtime/issues/116304")]
+    [ConditionalClass(typeof(MLKem), nameof(MLKem.IsSupported))]
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public sealed class MLKemCngExportableTests : MLKemCngTests
+    {
+        protected override CngExportPolicies ExportPolicies => CngExportPolicies.AllowExport;
+    }
+
+    [ConditionalClass(typeof(MLKem), nameof(MLKem.IsSupported))]
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public static class MLKemCngNonExportableTests
+    {
+        [Theory]
+        [MemberData(nameof(MLKemTestData.MLKemAlgorithms), MemberType = typeof(MLKemTestData))]
+        public static void MLKemCng_NonExportable_ExportPrivateSeedThrows(MLKemAlgorithm algorithm)
+        {
+            using CngKey key = MLKemCngTests.GenerateCngKey(algorithm, CngExportPolicies.None);
+            using MLKemCng kem = new MLKemCng(key);
+            Assert.Throws<CryptographicException>(() => kem.ExportPrivateSeed());
+        }
+
+        [Theory]
+        [MemberData(nameof(MLKemTestData.MLKemAlgorithms), MemberType = typeof(MLKemTestData))]
+        public static void MLKemCng_NonExportable_ExportDecapsulationKeyThrows(MLKemAlgorithm algorithm)
+        {
+            using CngKey key = MLKemCngTests.GenerateCngKey(algorithm, CngExportPolicies.None);
+            using MLKemCng kem = new MLKemCng(key);
+            Assert.Throws<CryptographicException>(() => kem.ExportDecapsulationKey());
+        }
+
+        [Fact]
+        public static void MLKemCng_NonExportable_ExportEncapsulationKeyAlwaysWorks()
+        {
+            using CngKey key = MLKemCngTests.ImportPrivateSeed(
+                MLKemAlgorithm.MLKem512,
+                MLKemTestData.MLKem512PrivateSeed,
+                CngExportPolicies.None);
+
+            using MLKemCng kem = new MLKemCng(key);
+            byte[] exportedKey = kem.ExportEncapsulationKey();
+            AssertExtensions.SequenceEqual(MLKemTestData.MLKem512EncapsulationKey, exportedKey);
+        }
+    }
+
+    [ConditionalClass(typeof(MLKem), nameof(MLKem.IsSupported))]
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public static class MLKemCngContractTests
+    {
+        [Fact]
+        public static void MLKemCng_Ctor_ArgValidation()
```

Comment:
This won't be run on non-Windows and on Windows without ML-KEM support, even though the MLKemCng constructor is accessible in both cases.

Reply:
The way `MLKemCng` (and `MLDsaCng`) are constructed is the PNSE happens before the arg-not-null check happens.

This will fail on macOS:

```C#
AssertExtensions.Throws<ArgumentNullException>("key", static () => new MLDsaCng(null));
```

I can fix it, but I was keeping consistent with that MLDsaCng did.

Reply:
I think the behavior is fine, I was just suggesting adding tests for those other conditions.

Reply:
I will address any actionable concerns in a follow up PR if needed, but will merge this for now.